### PR TITLE
Tejas ex 1-3

### PR DESCRIPTION
**Pull request by tejaskale**
Some changes and improvements from my side. 
Still need to look at Ex 03. So WIP!

### CODE REVIEW

**Comment by tejaskale** [comment]
Well , gitlab style WIP tags dont work on github :disappointed: 

Dont merge yet.

File: 2/art/art.py [hunk 1]
```diff
@@ -23,6 +24,32 @@ def art(A, b, iterations):
             if sqrd_norm <= 0:
                 continue
             else:
-                x += (b[j][0] - np.dot(row, x)) * np.conjugate(row) / sqrd_norm
+                x += relax * ((b[j][0] - np.dot(row, x)) * np.conjugate(row) / sqrd_norm)
```
Comment (momenator) [inline 2/art/art.py:27]: What is this relaxation factor?

File: 2/art/art.py [hunk 2]
```diff
@@ -23,6 +24,32 @@ def art(A, b, iterations):
...
             
     return x
+
+def art_anim(A, b, iterations, relax):
```
Comment (momenator) [inline 2/art/art.py:31]: What does  this function do exactly?

File: 2/art/main.py [hunk 3]
```diff
@@ -69,11 +69,11 @@
 
 # Solve LSE with ART
 start_time = timeit.default_timer()
-x_art = art(A, b, iterations)
+x_art = art(A, b, iterations,1)
```
Comment (momenator) [inline 2/art/main.py]: If relax is 1, then it would not make a difference wouldn't it?

**Review by momenator — CHANGES_REQUESTED** [review]

File: 2/art/art.py (hunk 1 above)
Comment (tejaskale) [reply 2/art/art.py:27]: I was just experimenting with the `lambda` from the wikipedia article: https://en.wikipedia.org/wiki/Algebraic_reconstruction_technique

File: 2/art/art.py (hunk 2 above)
Comment (tejaskale) [reply 2/art/art.py:31]: Animates each iteration of ART.

File: 2/art/main.py (hunk 3 above)
Comment (tejaskale) [reply 2/art/main.py]: Didnt want to break already working code

**Review by momenator — DISMISSED** [review]

**Comment by momenator** [comment]
Looks good, if you don't have any more things to add, these things are good to merge.

**Comment by tejaskale** [comment]
Check the third exercise. Looks like just the prints were the issue.

**Review by momenator — APPROVED** [review]

**Pull request merged**